File: CSharp/Lab2_1-2_2/Lab2_2/Circle.cs
namespace Lab2_2;

public class Circle : Point
{
    protected double MinorAxis;
    public double Radius
    {
        get => MinorAxis;
        protected set => MinorAxis = value;
    }

    public Circle(double x, double y, double radius) : base(x, y)
    {
        this.Radius = Math.Abs(radius);
        if (radius == 0.0) throw new ArgumentException("Коло не може бути точкою");
    }

    public double Area
    {
        get => Math.PI * (this.Radius * this.Radius);
    }
    
}

File: Cpp/Lab2_1-2_2/Lab2_2/Circle.cpp
#include <stdexcept>
#include <cmath>
#include "Circle.h"

Circle::Circle(double x, double y, double radius) : Point(x, y), radius(std::abs(radius)) {
    if(radius == 0.0) throw std::invalid_argument("Коло не може бути точкою");
}

double Circle::area() {
    return M_PI * (this->radius*this->radius);
}
